core/esmf-aspect-model-aas-generator/src/main/java/org/eclipse/esmf/aspectmodel/aas/AasToAspectModelGenerator.java

```diff
@@ -187,13 +181,29 @@ private <T> Collector<T, ArrayDeque<T>, ArrayDeque<T>> reverseOrder() {
    }
 
    private String iriToReversedHostNameNotation( final IRI iri ) {
-      final URI uri = URI.create( iri.toString().contains( "://" ) ? iri.toString() : "https://" + iri );
-      return Stream.concat(
-                  Arrays.stream( uri.getHost().split( "\\." ) ).collect( reverseOrder() ).stream(),
-                  Arrays.stream( uri.getPath().split( "/" ) ) )
+      URI uri;
+      try {
+         uri = URI.create( iri.toString().contains( "://" ) ? iri.toString() : "https://" + iri );
+      } catch ( IllegalArgumentException e ) {
+         throw new IllegalArgumentException( "Incorrect IRI: " + iri, e );
+      }
+
+      if ( uri.getHost() == null ) {
+         throw new IllegalArgumentException( "URI doesn't contain host: " + uri );
+      }
+
+      final String[] hostParts = uri.getHost().split( "\\." );
+      final List<String> hostPartsList = Arrays.asList( hostParts );
+      Collections.reverse( hostPartsList );
+      final String reversedHost = String.join( ".", hostPartsList );
+
+      final String[] pathParts = uri.getPath().split( "/" );
+      final String path = Arrays.stream( pathParts )
             .filter( StringUtils::isNotBlank )
-            .map( this::escapeUrnNamespacePart )
+            .limit( Math.max( 0, pathParts.length - 2 ) )
```

Comment:
This doesn't seem right. Why would we want to limit the length of the parts of a namespace main part (the reverse host name notation part of the Aspect Model URN)? Or am I mistaking the context?

Reply:
Yes, you are totally right. 
Here, I wanted to skip numbers in the end of main part of urn. But you are totally right.

Thanks!